DOC, CI: Docs build is failing on CircleCI

See, for example, https://app.circleci.com/pipelines/github/numpy/numpy/27407/workflows/01083f16-6603-4391-a31a-d4c839924e7f/jobs/40978 or https://app.circleci.com/pipelines/github/numpy/numpy/27412/workflows/035d8506-affe-46f1-8467-249f6c6856e3/jobs/40983?invite=true#step-108-5389_21. 

The error message is as follows.

<details>

```
<snip>
Running Sphinx v7.2.6

Traceback (most recent call last):
  File "/home/circleci/repo/venv/lib/python3.11/site-packages/sphinx/config.py", line 358, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/home/circleci/repo/doc/source/conf.py", line 65, in <module>
    replace_scalar_type_names()
  File "/home/circleci/repo/doc/source/conf.py", line 45, in replace_scalar_type_names
    assert 'numpy._core._add_newdocs_scalars' not in sys.modules
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/circleci/repo/venv/lib/python3.11/site-packages/sphinx/cmd/build.py", line 293, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/circleci/repo/venv/lib/python3.11/site-packages/sphinx/application.py", line 211, in __init__
    self.config = Config.read(self.confdir, confoverrides or {}, self.tags)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/circleci/repo/venv/lib/python3.11/site-packages/sphinx/config.py", line 181, in read
    namespace = eval_config_file(filename, tags)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/circleci/repo/venv/lib/python3.11/site-packages/sphinx/config.py", line 371, in eval_config_file
    raise ConfigError(msg % traceback.format_exc()) from exc
sphinx.errors.ConfigError: There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/home/circleci/repo/venv/lib/python3.11/site-packages/sphinx/config.py", line 358, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/home/circleci/repo/doc/source/conf.py", line 65, in <module>
    replace_scalar_type_names()
  File "/home/circleci/repo/doc/source/conf.py", line 45, in replace_scalar_type_names
    assert 'numpy._core._add_newdocs_scalars' not in sys.modules
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError


Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/home/circleci/repo/venv/lib/python3.11/site-packages/sphinx/config.py", line 358, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/home/circleci/repo/doc/source/conf.py", line 65, in <module>
    replace_scalar_type_names()
  File "/home/circleci/repo/doc/source/conf.py", line 45, in replace_scalar_type_names
    assert 'numpy._core._add_newdocs_scalars' not in sys.modules
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

make: *** [Makefile:144: html-build] Error 2

Exited with code exit status 2
```
</details>

My guess .circleci/config.yml needs to be updated. I'll look into it

--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -41,10 +41,6 @@ def replace_scalar_type_names():
         ('tp_name', ctypes.c_char_p),
     ]
 
-    # prevent numpy attaching docstrings to the scalar types
-    assert 'numpy._core._add_newdocs_scalars' not in sys.modules
-    sys.modules['numpy._core._add_newdocs_scalars'] = object()
-
     import numpy
 
     # change the __name__ of the scalar types
@@ -58,9 +54,6 @@ def replace_scalar_type_names():
         c_typ = PyTypeObject.from_address(id(typ))
         c_typ.tp_name = _name_cache[typ] = b"numpy." + name.encode('utf8')
 
-    # now generate the docstrings as usual
-    del sys.modules['numpy._core._add_newdocs_scalars']
-    import numpy._core._add_newdocs_scalars
 
 replace_scalar_type_names()
 

